Feat/emote preview controller

juanmahidalgo opened the pull request:
Closes #2159

![image](https://user-images.githubusercontent.com/8763687/180426749-7d04c13a-b9a7-4ad2-9e1b-c696f856abb7.png)

https://user-images.githubusercontent.com/8763687/180426770-a16f153a-e8bf-4fcd-ba95-f66000d41bfe.mov

LautaroPetaccio commented on src/components/Modals/CreateSingleItemModal/EditThumbnailStep/EditThumbnailStep.tsx:
```diff
@@ -0,0 +1,223 @@
+    }
+
+    let counter = currentFrame || 0
+    let max = length * 100
```
What's length defined in? What if we navigate the frames by a fixed adjustable percentage?

juanmahidalgo replied:
It's defined in seconds. Would you mind putting an example of how to navigate it with a fixed adjustable %?

LautaroPetaccio commented on src/components/Modals/CreateSingleItemModal/EditThumbnailStep/EditThumbnailStep.tsx:
```diff
@@ -0,0 +1,223 @@
+    let counter = currentFrame || 0
+    let max = length * 100
+    let intervalId: NodeJS.Timer
+    intervalId = setInterval(() => {
```
Can't we communicate with the iframe to either be notified when some frames are advanced or at least to ask on which frame are we before setting it?

juanmahidalgo replied:
I believe it's not part of the [current API](https://github.com/decentraland/wearable-preview#controller-rpc), it needs to be track in this component.

LautaroPetaccio reviewed (changes requested):
Great job! 

I’ve seen the following issues with the component (that I'm not sure if they're on the scope of the PR):
1. The zoom out, zoom in features restart the animation, but zooming in and out using the mouse, don’t.
2. Zooming the figure 3 or 4 times removes the model, or is so zoomed out that it’s not possible to see the image.
3. We should have a maximum amount of zoom in and out, as it can result in a zoom so big that the model gets cut.
4. The interval bar ends way sooner than the animation.
    a) When clicking the restart button, the model is not waited until loaded but the slider is still going.
    b) The slider seems to always stop at the end, lacking a bit of the slide and then finishes.
5. This is just a question, is the play button the same as the mocks? The black background seems a bit off in comparison with the UI.
6. The saved thumbnail has a black background, making it difficult to see.
7. The UI lets you choose a thumbnail but then overrides it with the one chosen in the next path.
8. The jumps in the slider are a bit rough (or big), maybe using smaller steps could be better?
9. I was expecting that the thumbnail picture was going to open or switch the modal to the thumbnail creation view so after choosing the thumbnail I could go back to make a final review o the item with its thumbnail.
10. After finishing the animation, the avatar spins, is this expected?
11. The model and the background are both pretty dark, wouldn’t it be better if we made it clearer?

juanmahidalgo commented on src/modules/translation/languages/es.json:
```diff
@@ -196,6 +196,7 @@
   },
   "create_single_item_modal": {
     "title": "Nuevo item",
+    "thumbnail_step_title": "Editar el thumbnail",
```
I used `thumbnail` because is not translated in our spanish keys at the moment:
![image](https://user-images.githubusercontent.com/8763687/180718057-2dd92176-edb0-4112-9e9f-1861df8f400a.png)

![image](https://user-images.githubusercontent.com/8763687/180718079-f05a4bde-0240-45c5-b5bd-07db81e999ae.png)

juanmahidalgo commented on src/components/Modals/CreateSingleItemModal/utils.ts:
```diff
@@ -28,3 +31,35 @@ export function validateEnum(name: keyof ItemAssetJson, assetJson: ItemAssetJson
   }
 }
+
+export function toWearableWithBlobs(file: File, isEmote = false): WearableWithBlobs {
```
I _believe_ it can be used for wearables as well, it can turn any model to a `WearableWithBlobs` type. I picked the function from here: https://github.com/decentraland/ui/blob/master/src/components/WearablePreview/WearablePreview.stories.tsx#L43.

juanmahidalgo commented:
> Great job!
> 
> I’ve seen the following issues with the component (that I'm not sure if they're on the scope of the PR):
> 
> 1. The zoom out, zoom in features restart the animation, but zooming in and out using the mouse, don’t.
> 2. Zooming the figure 3 or 4 times removes the model, or is so zoomed out that it’s not possible to see the image.
> 3. We should have a maximum amount of zoom in and out, as it can result in a zoom so big that the model gets cut.
> 4. The interval bar ends way sooner than the animation.
>    a) When clicking the restart button, the model is not waited until loaded but the slider is still going.
>    b) The slider seems to always stop at the end, lacking a bit of the slide and then finishes.
> 5. This is just a question, is the play button the same as the mocks? The black background seems a bit off in comparison with the UI.
> 6. The saved thumbnail has a black background, making it difficult to see.
> 7. The UI lets you choose a thumbnail but then overrides it with the one chosen in the next path.
> 8. The jumps in the slider are a bit rough (or big), maybe using smaller steps could be better?
> 9. I was expecting that the thumbnail picture was going to open or switch the modal to the thumbnail creation view so after choosing the thumbnail I could go back to make a final review o the item with its thumbnail.
> 10. After finishing the animation, the avatar spins, is this expected?
> 11. The model and the background are both pretty dark, wouldn’t it be better if we made it clearer?

1. That's a known issue, as I commented before, sending messages to the iframe restarts the animation. I believe we can keep it that way for now and try to fix it later.
2. Fixed with a MIN & MAX zoom values.
3. MIN & MAX zoom values added.
4. I believe that depends on the animation trying out. I changed the Icon because it wasn't clear, it's not a _restart_ one, it's to trigger the auto-rotation. I believe now it's a bit clearer. Clicking it sends a message to the iframe, making it restart the animation.
5. It's the same as the mocks
6. Background changed to be more lighter, I think it's better now
7. Sorry, can you be more specific? What's the next path? 
8.  I changed the step to be smaller, let me know if it's smoother now!
9. According the mocks, after picking the thumbnail, the next step is setting the price and then it's just saving it. I haven't seen a review step. We can discuss that with the UX team.
10. Haven't seen the spin, could you add a small video or something please? 
11. Changed the background to be a bit lighter, I think it's better now.

juanmahidalgo commented on src/components/Modals/CreateSingleItemModal/EditThumbnailStep/EditThumbnailStep.css:
```diff
@@ -0,0 +1,100 @@
+  margin-top: -4px;
+}
+
+.EditThumbnailStep input[type='range']::-webkit-slider-runnable-track {
```
Added support for firefox & microsoft's browsers.

juanmahidalgo commented on src/components/Modals/CreateSingleItemModal/EditThumbnailStep/EditThumbnailStep.css:
```diff
@@ -0,0 +1,100 @@
+}
+
+.EditThumbnailStep input[type='range'] {
+  -webkit-appearance: none;
```
It just won't be apply. I added support for firefox now

juanmahidalgo commented on src/components/Modals/CreateSingleItemModal/EditThumbnailStep/EditThumbnailStep.tsx:
```diff
@@ -0,0 +1,223 @@
+      counter += 50
+      const nextValue = counter >= max ? max : counter
+      this.setState({ frame: nextValue, isPlaying: nextValue === max ? false : true, playingIntervalId: intervalId })
+    }, 500)
```
Would you mind explaining the benefit of that change? I think I'm missing it
